glambie/processing/processing_helpers.py

```diff
@@ -407,59 +407,59 @@ def prepare_seasonal_calibration_dataset(region_config: RegionRunConfig,
         Extended annual timeseries.
-        If trends are within 'annual_timeseries' this will be the same as 'annual_timeseries'
+        If the desired time window is within 'annual_timeseries' and 'annual_timeseries' does not have any data gaps
+        this will be the same as 'annual_timeseries'
     """
     annual_timeseries_copy = annual_timeseries.copy()
-    for ds in data_catalogue_trends.datasets:
-        if (min(ds.data.start_dates) < min(annual_timeseries_copy.data.start_dates)) \
-                or (max(ds.data.end_dates) > max(annual_timeseries_copy.data.end_dates)) \
-                or not annual_timeseries_copy.data.is_cumulative_valid():  # or the case where the timeseries has a gap
-            log.info("Extension of annual is performed, as the trends are longer than the annual timeseries")
-
-            # Remove trend of timeseries for extension over the common time period
-            catalogue_dfs = [annual_timeseries_copy.data.as_dataframe(), timeseries_for_extension.data.as_dataframe()]
-            start_ref_period = np.max([df.start_dates.min() for df in catalogue_dfs])
-            end_ref_period = np.min([df.end_dates.max() for df in catalogue_dfs])
-            if not start_ref_period < end_ref_period:
-                warnings.warn("Warning when removing trends. No common period detected.")
-            for df in catalogue_dfs:
-                df_sub = df[(df["start_dates"] >= start_ref_period) & (df["end_dates"] <= end_ref_period)]
-                df["changes"] = df["changes"] - df_sub["changes"].mean()  # edit the dataframe
-
-            # Combine with other timeseries to cover the missing timespan
-            df_merged = pd.merge(catalogue_dfs[0], catalogue_dfs[1],
-                                 on=["start_dates", "end_dates"], how="outer")
-            # Fill Nans in 'annual_timeseries' with values from 'timeseries_for_extension'
-            df_merged.changes_x.fillna(df_merged.changes_y, inplace=True)
-            df_merged.errors_x.fillna(df_merged.errors_y, inplace=True)
-            df_merged = df_merged.sort_values(by="start_dates").reset_index()
-            # now update the annual timeseries object with the extended timeseries
-            annual_timeseries_copy.data.changes = np.array(df_merged["changes_x"])
-            annual_timeseries_copy.data.errors = np.array(df_merged["errors_x"])
-            annual_timeseries_copy.data.start_dates = np.array(df_merged["start_dates"])
-            annual_timeseries_copy.data.end_dates = np.array(df_merged["end_dates"])
-            return annual_timeseries_copy
+    if (desired_time_window[0] < min(annual_timeseries_copy.data.start_dates)) \
+            or (desired_time_window[1] > max(annual_timeseries_copy.data.end_dates)) \
+            or not annual_timeseries_copy.data.is_cumulative_valid():  # or the case where the timeseries has a gap
+        log.info("Extension of annual is performed, as the trends are longer than the annual timeseries")
```

Comment:
The grammar here is strange: can we avoid using Passive Voice in logging statements please? Ideally the statement should make it clear whether you are _about to_ do something or _have already done_ something.

Reply:
Done